### PR TITLE
Fim puppet addition

### PR DESCRIPTION
**Pull request by ch7ck**
Add file monitoring

### CODE REVIEW

**Comment by ch7ck** [comment]
These commits include the changes in pull request #18

File: manifests/fileim.pp [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# == Class opsmatic::fileim
+#
+# Places file-integrity-monitoring JSON file in /var/db/opsmatic-agent/external.d/
+#
+# === Authors
+#
+# Opsmatic Inc. (support@opsmatic.com)
+#
+class opsmatic::fileim{
```
Comment (mihasya) [inline manifests/fileim.pp]: not sure I'm crazy about the name `fileim`. How about `monitor_files`?

File: spec/classes/agent_spec.rb [hunk 2]
```diff
@@ -23,18 +23,17 @@
     end
   end
 
-  context 'token => 1234, files_config_enabled => true' do
+  context 'token => 1234' do
     let(:facts) { FACTS }
     let(:params) {{
-      :token => '1234', :files_config_enabled => true
-    }}
+      :token => '1234'    }}
```
Comment (mihasya) [inline spec/classes/agent_spec.rb]: formatting looks weird

File: manifests/agent.pp [hunk 3]
```diff
@@ -18,13 +18,12 @@
 class opsmatic::agent (
   $ensure               = $opsmatic::params::agent_ensure,
   $token                = $opsmatic::params::token,
-  $files_config_enabled = $opsmatic::params::files_config_enabled,
+  $host_alias           = $opsmatic::params::host_alias,
+  $filemonitorlist      = $opsmatic::params::filemonitorlist,
```
Comment (mihasya) [inline manifests/agent.pp]: should we also pull through `$groups`

File: spec/classes/agent_spec.rb [hunk 4]
```diff
@@ -23,18 +23,17 @@
...
     it do
       should compile.with_all_deps
       should contain_class('opsmatic::debian')
       should contain_package('opsmatic-agent').with(
         'ensure' => 'present')
       should contain_file('/etc/opsmatic-agent.conf').with(
-        'content' => /paths_ignore = \[\]\nfiles_config_enabled = true\ngroups = ""/)
+        'content' => /host_alias = ""\ngroups = ""/)
```
Comment (mihasya) [inline spec/classes/agent_spec.rb]: if we're going to include `host_alias` and `groups` in this PR, we should make the tests check that they work

File: manifests/agent.pp [hunk 5]
```diff
@@ -218,8 +216,10 @@
           fail('Opsmatic Agent is not supported on this platform')
         }
       }
+
     }
     default: {
+      include opsmatic::monitor_files
```
Comment (mihasya) [inline manifests/agent.pp]: why do we include monitor_files in the block which appears to uninstall the agent?

File: manifests/agent.pp (hunk 3 above)
Comment (ch7ck) [inline manifests/agent.pp]: groups is not used by the agent manifest.

It gets set by the template-erb -> https://github.com/opsmatic/puppet-opsmatic/blob/master/templates/opsmatic-agent.conf.erb

File: spec/classes/agent_spec.rb (hunk 2 above)
Comment (ch7ck) [inline spec/classes/agent_spec.rb]: This gives some sort of token....the test needs to be rewritten or scrapped (made an issue for it)

https://github.com/opsmatic/puppet-opsmatic/issues/20

File: spec/classes/agent_spec.rb (hunk 4 above)
Comment (ch7ck) [inline spec/classes/agent_spec.rb]: +1 https://github.com/opsmatic/puppet-opsmatic/issues/20

**Pull request merged**